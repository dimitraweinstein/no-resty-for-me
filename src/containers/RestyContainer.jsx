import React, { Component } from 'react';
import Controls from '../components/Controls';
import History from '../components/History';
import Display from '../components/Display';
import { fetchApi } from '../services/restyFetchApi';


export default class RestyContainer extends Component {
  state = {
    loading: false,
    url: '',
    body: '',
    jsonResponse: {},
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    this.setState({ loading: true });
    //fetch data function here 
    const jsonResponse = await fetchApi(this.state.url);
    //set fetched data to state and set loading state to false
    this.setState({ jsonResponse, loading: false });
  }

  render() {
    const { loading, url, body, jsonResponse } = this.state;
    return (
      <>
        <Controls
          url={url}
          body={body}
          onChange={this.handleChange}
          onSubmit={this.handleSubmit}
        />

        { loading ? (<img
          // eslint-disable-next-line max-len
          src="https://icon-library.com/images/ajax-loading-icon/ajax-loading-icon-2.jpg"
          alt="loading spinner" />) : (
          <>
            <History url={url} />
            <Display jsonResponse={jsonResponse} />
          </>
        )
        }
      </>
    );
  }
}
