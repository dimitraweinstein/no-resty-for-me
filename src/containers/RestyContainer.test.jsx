import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import RestyContainer from './RestyContainer';

describe('RESTy Container', () => {
  it('should render RESTy App components on page', async () => {
    render(<RestyContainer />);

    // eslint-disable-next-line max-len
    const urlInput = screen.getByPlaceholderText('URL', { placeholder: '' });
    expect(urlInput).toBeEmptyDOMElement();
    
    //test for radio buttons here
    const radioInput = await screen.findByDisplayValue('GET');
    userEvent.click(radioInput);

    //test raw json text area here
    //test for submit button here
    // const submit = screen.getByRole('button', { name: 'get-api' });
    // userEvent.click(submit);

    const ul = await screen.findByRole('list', { name: 'urls' });
    expect(ul).toBeInTheDocument();

    const div = await screen.findByRole('response', { name: 'response' });
    expect(div).toBeInTheDocument();

  });
});
