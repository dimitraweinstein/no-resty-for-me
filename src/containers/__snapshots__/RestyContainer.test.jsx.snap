// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RESTy Container should render RESTy App components on page 1`] = `
<div
  aria-label="response"
  class="response-div"
  role="response"
>
  <div
    class="react-json-view"
    style="font-family: monospace; cursor: default; background-color: rgba(0, 0, 0, 0); position: relative;"
  >
    <div
      class="pretty-json-container object-container"
    >
      <div
        class="object-content"
      >
        <div
          class="object-key-val"
        >
          <span>
            <span
              style="display: inline-block; cursor: pointer;"
            >
              <div
                class="icon-container"
                style="display: inline-block; width: 17px;"
              >
                <span
                  class="expanded-icon"
                >
                  <svg
                    fill="currentColor"
                    style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                    viewBox="0 0 15 15"
                  >
                    <path
                      d="M0 5l6 6 6-6z"
                    />
                  </svg>
                </span>
              </div>
              <span
                style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
              >
                <span
                  class="object-key"
                >
                  <span
                    style="vertical-align: top;"
                  >
                    "
                  </span>
                  <span>
                    root
                  </span>
                  <span
                    style="vertical-align: top;"
                  >
                    "
                  </span>
                </span>
                <span
                  style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                >
                  :
                </span>
              </span>
              <span
                style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
              >
                {
              </span>
            </span>
            <div
              class="object-meta-data"
              style="display: inline-block; padding: 0px 0px 0px 10px;"
            >
              <span
                class="object-size"
                style="color: rgba(0, 0, 0, 0.3); border-radius: 3px; font-style: italic; margin: 0px 6px 0px 0px; cursor: default;"
              >
                1
                 item
                
              </span>
            </div>
          </span>
          <div
            class="pushed-content object-container"
          >
            <div
              class="object-content"
              style="margin-left: 6px;"
            >
              <div
                class="variable-row"
                style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
              >
                <span>
                  <span
                    class="object-key"
                    style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                  >
                    <span
                      style="vertical-align: top;"
                    >
                      "
                    </span>
                    <span
                      style="display: inline-block;"
                    >
                      Fetch Me!
                    </span>
                    <span
                      style="vertical-align: top;"
                    >
                      "
                    </span>
                  </span>
                  <span
                    style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                  >
                    :
                  </span>
                </span>
                <div
                  class="variable-value"
                  style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                >
                  <div
                    style="display: inline-block; color: rgb(203, 75, 22);"
                  >
                    <span
                      class="string-value"
                      style="cursor: default;"
                    >
                      "
                      Fetch Me Right Now!
                      "
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <span
            class="brace-row"
          >
            <span
              style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
            >
              }
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
